binding/form_mapping.go

```diff
@@ -170,6 +170,11 @@ func setByForm(value reflect.Value, field reflect.StructField, form map[string][
 }
 
 func setWithProperType(val string, value reflect.Value, field reflect.StructField) error {
+	// If it is a string type, no spaces are removed, and the user data is not modified here
+	if value.Kind() != reflect.String {
+		val = strings.TrimSpace(val)
```

Comment:
If `val` is not string type, can call `func TrimSpace(s string) string`?

Reply:
The type of val is string, as explained below.
If the client command is the following code

```curl -H 'intkey: v1" -H "strkey: v2" url ```
We define the following structure to parse the http header
```go
type needHeader struct {
    IntKey int `header:"intkey"
    StrKey string `header:"strkey"
}
```
Such a data structure.
Before setting into the structure, regardless of the int or string required by the structure, it is of type string before being parsed.